src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -1128,6 +1125,15 @@ async def extend(self, data):
             )
         return data
 
+    @private
+    async def extend_context(self, extra):
+        return {
+            'groups': await self.middleware.call(
```

Comment:
Do you think we should maybe have this normalized here so that we don't pop entries from groups each time at L1113 ?

Reply:
Nice catch, thank you!